packages/grid/_modules_/grid/components/styled-wrappers/GridToolbar.tsx

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import { classnames } from '../../utils';
+
+type GridToolbarProps = React.HTMLAttributes<HTMLDivElement>;
+
+export const GridToolbar = React.forwardRef<HTMLDivElement, GridToolbarProps>(function GridToolbar(
+  props,
+  ref,
+) {
+  const { className, children, ...other } = props;
+
+  if (!children) {
+    return null;
+  }
+
+  return (
+    <div ref={ref} className={classnames('MuiDataGrid-toolbar', className)} {...other}>
+      {children}
+    </div>
+  );
```

Comment:
This behavior would break the standard rendering logic of the core components. Why render null if no children are provided?

```suggestion
  const { className, ...other } = props;

  return (<div ref={ref} className={classnames('MuiDataGrid-toolbar', className)} {...other} />);
```

Reply:
> Why would somebody use the `toolbar` over the `header`?

Idk, I can move it under the header. Now that I think about it it does sound logical to be under it.

Reply:
Yeah, sounds better, no need for it;

Reply:
> This behavior would break the standard rendering logic of the core components. Why render null if no children are provided?

This is a temporary logic. It should be removed once there at least 1 thing in the toolbar. If I remove this check now the toolbar will render and it will be empty, It will only take space. I did it that way so that if we want to release next Tuesday lets say and the `ColumnPicker` or `DencitySelector` or the filtering are not yet ready or merged we can still do it without introducing an empty toolbar that is aways visible by default.

Reply:
> This is a temporary logic. It should be removed once there at least 1 thing in the toolbar.

The more I think about it, the more I think this is the wrong direction, even for the short term. Developers will want to decide which action in the toolbar they display. How will we do that? With a boolean for each feature? A composable API sounds simpler

Reply:
Check this one: https://github.com/mui-org/material-ui-x/issues/575.

Reply:
I agree the better approach would have been to have it as a separate component wrapped around a provider but we chose to do it like this. Also from our discussions, I didn't expect to have the toolbar inside the Grid for a long time but more of a temporary solution to speed up the feature development.

But yes you are right it is bad to have a boolean per feature to disable them. For the current implementation I was more thinking in terms of - if you don't want certain features, provide your own toolbar because otherwise, the public API will be just too big and badly designed.